plugins/jumpcloud/pkg/fetch/fetch.go

```diff
@@ -60,45 +60,57 @@ func (f *Fetcher) Fetch(ctx context.Context, outputWriter, errorWriter io.Writer
 	}
 
 	if f.Groups {
-		groups, err := f.jcc.ListGroups(ctx, jc.UserGroups)
+		if err := f.fetchGroups(ctx, writer, errorWriter, idLookup); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (f *Fetcher) fetchGroups(ctx context.Context,
+	writer *js.JSONArrayWriter,
+	errorWriter io.Writer,
+	idLookup map[string]*jc.BaseUser,
+) error {
+	groups, err := f.jcc.ListGroups(ctx, jc.UserGroups)
+	if err != nil {
+		common.WriteErrorWithExitCode(errorWriter, err, 1)
+		return err
+	}
+
+	for _, group := range groups {
+		groupBytes, err := json.Marshal(group)
 		if err != nil {
 			common.WriteErrorWithExitCode(errorWriter, err, 1)
-			return err
+			continue
 		}
 
-		for _, group := range groups {
-			groupBytes, err := json.Marshal(group)
-			if err != nil {
-				common.WriteErrorWithExitCode(errorWriter, err, 1)
-				continue
-			}
+		var obj map[string]interface{}
+		if err := json.Unmarshal(groupBytes, &obj); err != nil {
+			common.WriteErrorWithExitCode(errorWriter, err, 1)
+			continue
+		}
 
-			var obj map[string]interface{}
-			if err := json.Unmarshal(groupBytes, &obj); err != nil {
-				common.WriteErrorWithExitCode(errorWriter, err, 1)
-				continue
-			}
+		usersInGroup, err := f.jcc.ExpandUsersInGroup(ctx, group.ID, idLookup)
+		if err != nil {
+			common.WriteErrorWithExitCode(errorWriter, err, 1)
+		}
 
-			usersInGroup, err := f.jcc.ExpandUsersInGroup(ctx, group.ID, idLookup)
-			if err != nil {
+		usersInGroupBytes, err := json.Marshal(usersInGroup)
+		if err != nil {
+			common.WriteErrorWithExitCode(errorWriter, err, 1)
```

Comment:
If `common.WriteErrorWithExitCode` calls `os.Exit` then the `else` is redundant and needlessly increases indentation.

Reply:
This does not actually call os.Exit

Reply:
But the third argument is still redundant because the only value ever used is `1`.
So this could be just `WriteError` and ideally, it would be a function on `errorWriter` so we could simply call
```go
errorWriter.Error(err)
```